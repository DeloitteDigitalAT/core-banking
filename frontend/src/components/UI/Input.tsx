import React from "react";

interface InputType {
  label: string;
  placeholder: string;
  value: string;
  required: boolean;
  error: boolean;
  setInput: (el: string) => void;
}

const Input: React.FC<InputType> = properties => {
  const { label, placeholder, value, setInput, required, error } = properties;
  return (
    <label className="block py-2 mt-4">
      <span className="text-gray-700">
        {label}
        {required ? " *" : null}
      </span>
      <input
        className={`max-w-xl form-input mt-1 block w-full border rounded py-2 px-2 ${error ? "border-mango-600" : ""}`}
        placeholder={placeholder}
        value={value}
        onChange={el => setInput(el.target.value)}
      />
    </label>
  );
};

export default Input;
