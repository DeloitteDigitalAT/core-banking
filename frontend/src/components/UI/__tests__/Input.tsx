import React from "react";
import renderer from "react-test-renderer";
import Input from "../Input";

describe("UI/Input component", () => {
  test("snapshot renders", () => {
    const tree = renderer
      .create(
        <Input
          label="label"
          placeholder="placeholder"
          value="value"
          required
          error
          readonly={false}
          setInput={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
