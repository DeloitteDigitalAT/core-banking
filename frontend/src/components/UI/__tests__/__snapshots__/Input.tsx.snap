// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Input component snapshot renders 1`] = `
<label
  className="block mt-4"
>
  <span
    className="block uppercase tracking-wide text-xs font-semibold mb-2 text-gray-700"
  >
    label
     *
  </span>
  <input
    className="form-input mt-1 block w-full text-sm border rounded py-2 px-2 border-mango-600 "
    onChange={[Function]}
    placeholder="placeholder"
    readOnly={false}
    value="value"
  />
</label>
`;
